application/src/main/java/org/togetherjava/tjbot/commands/free/ChannelMonitor.java

```diff
@@ -0,0 +1,245 @@
+     * . So always add the status channel AFTER you have added all monitored channels for the guild,
+     * see {@link #addChannelToMonitor(long)}.
+     *
+     * @param channel the channel the status message must be displayed in
+     */
+    public void addChannelForStatus(@NotNull final TextChannel channel) {
+        postStatusInChannel.put(channel.getGuild().getIdLong(), channel.getIdLong());
+        updateStatusFor(channel.getGuild());
+    }
+
+    /**
+     * This method tests whether a guild id is configured for monitoring in the free command system.
+     * 
+     * @param guildId the id of the guild to test.
+     * @return {@code true} if the guild is configured in the system, {@code false} otherwise.
+     */
+    public boolean isMonitoringGuild(final long guildId) {
+        return postStatusInChannel.containsKey(guildId);
+    }
+
+    /**
+     * This method tests whether a channel id is configured for monitoring in the free command
+     * system.
+     * 
+     * @param channelId the id of the channel to test.
+     * @return {@code true} if the channel is configured in the system, {@code false} otherwise.
+     */
+    public boolean isMonitoringChannel(final long channelId) {
+        return channelsToMonitor.containsKey(channelId);
+    }
+
+    /**
+     * This is a delegation method to pass operations to {@link ChannelStatus}. see
+     * {@link ChannelStatus#isBusy()} for details.
+     *
+     * @param channelId the id for the channel to test.
+     * @return {@code true} if the channel is 'busy', false if the channel is 'free'.
+     */
+    public boolean isChannelBusy(final long channelId) {
+        return channelsToMonitor.get(channelId).isBusy();
+    }
+
+    /**
+     * This method tests if a channel is currently active by fetching the latest message and testing
+     * if it was posted more than an hour ago.
+     *
+     * @param channel the channel to test.
+     * @return {@code true} if the channel is inactive, false if it has received messages more
+     *         recently than an hour ago.
+     */
+    public boolean isChannelInactive(@NotNull final TextChannel channel) {
+        if (!channelsToMonitor.containsKey(channel.getIdLong())) {
+            throw new IllegalArgumentException(
+                    "Channel requested %s is not monitored by free channel"
+                        .formatted(channel.getName()));
+        }
+        // todo change the entire inactive test to work via restactions
+        return channel.getHistory()
+            .retrievePast(1)
+            .map(messages -> messages.get(0))
+            .map(Message::getTimeCreated)
+            .map(createdTime -> createdTime.isBefore(Util.anHourAgo()))
+            .complete();
+    }
+
+    /**
+     * This is a delegation method to pass operations to {@link ChannelStatus}. see
+     * {@link ChannelStatus#setBusy(long)} for details.
```

Comment:
Not delegating would break Law of Demeter. Currently 'FreeCommand' talks to 'ChannelMonitor' and 'ChannelMonitor' speaks to 'ChannelStatus'.

Without these delegation methods 'FreeCommand' would have to manually fetch instances of 'ChannelStatus' itself and modify them (and that modification will happen in multiple locations 'onSlashCommand' and 'onGuildMessageReceived' and any other additional functionality we add) rather than localised.